artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,347 @@
+	localJfrogHomePath string
+}
+
+// Creeate an InstallPluginCommand
+func NewInstallPluginCommand(artifactoryServerDetails *config.ServerDetails, pluginName string, fileTransferManger *PluginTransferManager) *InstallPluginCommand {
+	return &InstallPluginCommand{
+		targetServer:       artifactoryServerDetails,
+		transferManger:     fileTransferManger,
+		pluginName:         pluginName,
+		installVersion:     nil, // latest
+		localSrcDir:        "",
+		localJfrogHomePath: "",
+	}
+}
+
+func (tic *InstallPluginCommand) ServerDetails() (*config.ServerDetails, error) {
+	return tic.targetServer, nil
+}
+
+// Set the local directory that the plugin files will be copied from
+func (tic *InstallPluginCommand) SetLocalPluginFiles(localDir string) *InstallPluginCommand {
+	tic.localSrcDir = localDir
+	return tic
+}
+
+// Set the plugin version we want to download
+func (tic *InstallPluginCommand) SetInstallVersion(installVersion *version.Version) *InstallPluginCommand {
+	tic.installVersion = installVersion
+	return tic
+}
+
+// Set the base URL that the plugin files avaliable
+func (tic *InstallPluginCommand) SetBaseDownloadUrl(baseUrl string) *InstallPluginCommand {
+	tic.baseDownloadUrl = baseUrl
+	return tic
+}
+
+// Set the Jfrog home directory path override to search in it the plugin directory as destination
+func (tic *InstallPluginCommand) SetOverrideJfrogHomePath(path string) *InstallPluginCommand {
+	tic.localJfrogHomePath = path
+	return tic
+}
+
+// Validates minimum version and fetch the current
+func validateMinArtifactoryVersion(servicesManager artifactory.ArtifactoryServicesManager) (artifactoryVersion *version.Version, err error) {
+	log.Debug("verifying install requirements...")
+	rawVersion, err := servicesManager.GetVersion()
+	if err != nil {
+		return
+	}
+	artifactoryVersion = version.NewVersion(rawVersion)
+	if !artifactoryVersion.AtLeast(minArtifactoryVersion) {
+		err = minVerErr
+		return
+	}
+	return
+}
+
+// Get the plugin directory destination to install the plugin in it, search is starting from jfrog home directory
+func (tic *InstallPluginCommand) getPluginDirDestination() (target PluginFileItem, err error) {
+	var exists bool
+	var envVal string
+
+	// Flag override
+	if tic.localJfrogHomePath != "" {
+		log.Debug(fmt.Sprintf("try searching for plugin directory with custom Jfrog home directory '%s'.", tic.localJfrogHomePath))
+		if exists, target, err = tic.transferManger.trySearchDestinationMatchFrom(tic.localJfrogHomePath); err != nil || exists {
+			return
+		}
+	}
+	// Environment variable override
+	if envVal, exists = os.LookupEnv(jHomeEnvVar); exists {
+		log.Debug(fmt.Sprintf("try searching for plugin directory with '%s=%s'.", jHomeEnvVar, envVal))
+		if exists, target, err = tic.transferManger.trySearchDestinationMatchFrom(envVal); err != nil || exists {
+			return
+		}
+	}
+	// Default value
+	log.Debug(fmt.Sprintf("try searching for plugin directory with default path '%s'.", defalutSearchPath))
```

Comment:
`/opt/jfrog` can't exist in Windows - let's skip this in Windows.

Reply:
done